src/cladetime/util/sequence.py

```diff
@@ -1,86 +1,30 @@
 """Functions for retrieving and parsing SARS-CoV-2 virus genome data."""
 
-import json
 import lzma
-import zipfile
-from datetime import datetime, timezone
+import os
 from pathlib import Path
+from urllib.parse import urlparse
 
 import polars as pl
 import structlog
 import us
 from requests import Session
 
-from cladetime.util.reference import _get_s3_object_url
-from cladetime.util.session import _check_response, _get_session
+from cladetime.util.session import _get_session
 from cladetime.util.timing import time_function
 
 logger = structlog.get_logger()
 
 
 @time_function
-def get_covid_genome_data(released_since_date: str, base_url: str, filename: str):
-    """
-    Download genome data package from NCBI.
-    FIXME: Download the Nextclade-processed GenBank sequence data (which originates from NCBI)
-    from https://data.nextstrain.org/files/ncov/open/sequences.fasta.zst instead of using
-    the NCBI API.
-    """
-    headers = {
-        "Accept": "application/zip",
-    }
-    session = _get_session()
-    session.headers.update(headers)
-
-    # TODO: this might be a better as an item in the forthcoming config file
-    request_body = {
-        "released_since": released_since_date,
-        "taxon": "SARS-CoV-2",
-        "refseq_only": False,
-        "annotated_only": False,
-        "host": "Homo sapiens",
-        "complete_only": False,
-        "table_fields": ["unspecified"],
-        "include_sequence": ["GENOME"],
-        "aux_report": ["DATASET_REPORT"],
-        "format": "tsv",
-        "use_psg": False,
-    }
-
-    logger.info("NCBI API call starting", released_since_date=released_since_date)
-
-    response = session.post(base_url, data=json.dumps(request_body), timeout=(300, 300))
-    _check_response(response)
-
-    # Originally tried saving the NCBI package via a stream call and iter_content (to prevent potential
-    # memory issues that can arise when download large files). However, ran into an intermittent error:
-    # ChunkedEncodingError(ProtocolError('Response ended prematurely').
-    # We may need to revisit this at some point, depending on how much data we place to request via the
-    # API and what kind of machine the pipeline will run on.
-    with open(filename, "wb") as f:
-        f.write(response.content)
-
-
-@time_function
-def download_covid_genome_metadata(
-    session: Session, bucket: str, key: str, data_path: Path, as_of: str | None = None, use_existing: bool = False
-) -> Path:
-    """Download the latest GenBank genome metadata data from Nextstrain."""
+def _download_from_url(session: Session, url: str, data_path: Path) -> Path:
```

Comment:
Now that `CladeTime` provides the S3 URLs that point to the correct versions of sequence data and metadata, we can replace `download_covid_genome_metadata` with a more generic "download data using a given URL" function.